src/common/utils.cpp

```diff
@@ -47,22 +49,42 @@ static string ExtractIcebergScanPath(const string &sql) {
 	return sql.substr(start, end - start);
 }
 
+optional_ptr<CatalogEntry> IcebergUtils::GetICTableEntry(ClientContext &context, const string &input) {
+	auto qualified_name = QualifiedName::ParseComponents(input);
+	if (qualified_name.size() != 3) {
+		return nullptr;
+	}
+	EntryLookupInfo table_info(CatalogType::TABLE_ENTRY, qualified_name[2]);
+	auto table =
+	    Catalog::GetEntry(context, qualified_name[0], qualified_name[1], table_info, OnEntryNotFound::RETURN_NULL);
+	if (!table) {
+		throw InvalidInputException("Could not find Catalog Entry with name '%s'", input);
+	}
+	return table;
+}
+
+optional_ptr<ICTableEntry> IcebergUtils::GetIcebergTableEntry(ClientContext &context, const string &name) {
+	auto catalog_entry = IcebergUtils::GetICTableEntry(context, name);
+	if (catalog_entry->type == CatalogType::TABLE_ENTRY) {
+		//! This is a IRCTableEntry, set up the scan from this
```

Comment:
Correction, this is a `TableCatalogEntry` we need to cast it to that first, then get the `Catalog()` (maybe `ParentCatalog()`?) and check `GetCatalogType() == "iceberg"`